app/notification/model/notification_utils.py

```diff
@@ -1,59 +1,66 @@
 import re
 
+from flask import current_app
+
 
 def convert_bool_value(data):
-    def convert_values(value):
-        if value is None:
-            return "Not provided"
-        elif value is True:
-            return "Yes"
-        elif value is False:
-            return "No"
-        else:
-            return value
-
-    if isinstance(data, list):
-        if all(isinstance(sublist, list) for sublist in data):
-            converted_data = [
-                [convert_values(value) for value in sublist]
-                for sublist in data
-            ]
+    try:
+
+        def convert_values(value):
+            if value is None:
+                return "Not provided"
+            elif value is True:
+                return "Yes"
+            elif value is False:
+                return "No"
+            else:
+                return value
+
+        if isinstance(data, list):
+            if all(isinstance(sublist, list) for sublist in data):
+                converted_data = [
+                    [convert_values(value) for value in sublist]
+                    for sublist in data
+                ]
+            else:
+                converted_data = [convert_values(value) for value in data]
         else:
-            converted_data = [convert_values(value) for value in data]
-    else:
-        converted_data = convert_values(data)
+            converted_data = convert_values(data)
```

Comment:
I think this may be better as a recursive function:

https://github.com/communitiesuk/funding-service-design-notification/pull/81#discussion_r1219272781

Reply:
These two functions are not returning the same expected output in different scenarios 🤔 . Although I see your point on the Recursive call

Reply:
ah okay, sorry i didnt test that!